src/components/DownloadMailingListButton.tsx

```diff
@@ -1,52 +1,48 @@
-import { UnprocessedResultsFromCRM, OwnerType } from '../types'
+import { UnprocessedResultsFromCRM, IntersectedSearchAndFilterParams } from '../types'
 
 type DownloadButtonProps = {
+    searchParameters: IntersectedSearchAndFilterParams[];
     results: UnprocessedResultsFromCRM[]
 }
 export function DownloadMailingListButton (props: DownloadButtonProps) {
+    // const searchedAddress = '122 Parramatta Rd, Croydon NSW 2132, Australia'
+    const searchedAddress = props.searchParameters[0].searchAddress
     let downloadUrl = null
     const matchingPropertiesAndOwners = props.results
     const ownerContactDupeRemoval: string[] = []
     function generateCSVRow (propertyObject: UnprocessedResultsFromCRM) {
-        let csvRow = ''
+        let csvRowsForProperty = ''
         let doNotMail
         let returnToSender
         let postalAddress
+        const propertyReverseGeocodedAddress = propertyObject.Reverse_Geocoded_Address
         const propertyAddress = propertyObject.Deal_Name
         const propertyType = propertyObject.Property_Category_Mailing.join('; ')
-        const ownerObject = propertyObject.owner_details?.filter(item => item.Company)
-        const relatedContact = ownerObject?.find((owner: OwnerType) => owner.Contact_Type === 'Director')
-        const owner = ownerObject?.find((owner: OwnerType) => owner.Contact_Type === 'Owner')
+        const ownersArray = propertyObject.owner_details
 
-        let ownerOrRelatedContact = owner || relatedContact || null
-        if (!relatedContact && owner) {
-            doNotMail = owner.Do_Not_Mail
-            returnToSender = owner.Return_to_Sender
-            postalAddress = owner.Postal_Address
-            ownerOrRelatedContact = owner
-        } else if (relatedContact) {
-            doNotMail = relatedContact.Do_Not_Mail
-            returnToSender = relatedContact.Return_to_Sender
-            postalAddress = relatedContact.Postal_Address
-            ownerOrRelatedContact = relatedContact
-        }
-
-        const isDupe = ownerContactDupeRemoval.includes(`${ownerOrRelatedContact?.Postal_Address}-${ownerOrRelatedContact?.Name}`)
-        if (!doNotMail && !returnToSender) {
-            if (propertyAddress && postalAddress) {
-                if (!isDupe) {
-                    ownerContactDupeRemoval.push(`${ownerOrRelatedContact?.Postal_Address}-${ownerOrRelatedContact?.Name}`)
-                    const lastMailed = owner?.Last_Mailed || relatedContact?.Last_Mailed || 'Last mailed has not been found'
-                    csvRow = `"${ownerOrRelatedContact?.Name}","${ownerOrRelatedContact?.Contact_Type}","${postalAddress}","${ownerOrRelatedContact?.Postal_Suburb}","${ownerOrRelatedContact?.Postal_State}","${ownerOrRelatedContact?.Postal_Postcode}","${propertyAddress}",${propertyType},${lastMailed}\r\n`
-                    csvRow = csvRow.replace(/null/g, '-')
+        const resultIsSearchedAddress = propertyReverseGeocodedAddress.includes(searchedAddress)
```

Comment:
nitpick: preference to use is/has/should as a boolean prefix. e.g. isExactMatchForSearchAddress